### PR TITLE
Unconditionally build the cross-os DAC

### PR DESCRIPTION
**Pull request by hoyosjs**
Upload all variants of the cross OS DAC so it can be used in Runtime inner loop scenarios.

### CODE REVIEW

**Review by mikem8361 — APPROVED** [review]

**Comment by hoyosjs** [comment]
Hmm we have an uncaught `PAL_SEHException` in OSX while running the profiler eventpipe tests.

cc: @janvorli @dotnet/dotnet-diag

**Comment by janvorli** [comment]
> we have an uncaught `PAL_SEHException` in OSX while running the profiler eventpipe tests.

That means that this exception which represents managed exceptions was thrown from an unexpected place where we don't have the related macros to establish try / catch for it. However looking at the history, it looks like a one-off repro. It would require a local repro under a debugger or core dump from the lab to be diagnosable. I don't see a dump attached to that.

**Comment by hoyosjs** [comment]
> > we have an uncaught `PAL_SEHException` in OSX while running the profiler eventpipe tests.
> 
> That means that this exception which represents managed exceptions was thrown from an unexpected place where we don't have the related macros to establish try / catch for it. However looking at the history, it looks like a one-off repro. It would require a local repro under a debugger or core dump from the lab to be diagnosable. I don't see a dump attached to that.

There is one, I am trying to set it up.

**Comment by hoyosjs** [comment]
I'm merging this, since this has no product impact.

**Comment by sdmaclea** [comment]
Thanks @hoyosjs

**Comment by davmason** [comment]
@hoyosjs @janvorli I have seen the PAL_SEHException issue a few times before. I was thinking it was a one off previously, but it has happened enough that I think it warrants some amount of investigation.

Do you have advice on how to run the test in a loop under a debugger to catch it when it finally crashes? On windows I would create a script that runs the test in a loop and then run it under a debugger with child debugging enabled. I haven't had luck with that approach on OSX before

File: eng/pipelines/coreclr/templates/build-job.yml [hunk 1]
```diff
@@ -215,7 +215,7 @@ jobs:
           artifactName: $(buildProductArtifactName)
           displayName: 'product build'
 
-    - ${{ if and(ne(parameters.osGroup, 'OSX'), ne(parameters.archType, 'x86'), ne(parameters.compilerName, 'gcc'), ne(parameters.testGroup, 'clrTools')) }}:
+    - ${{ if and(in(parameters.osGroup, 'Windows_NT', 'Linux'), ne(parameters.archType, 'x86'), ne(parameters.compilerName, 'gcc'), ne(parameters.testGroup, 'clrTools')) }}:
```
Comment (safern) [inline eng/pipelines/coreclr/templates/build-job.yml:218]: How is this condition different than from what it was.

Comment (hoyosjs) [reply eng/pipelines/coreclr/templates/build-job.yml:218]: BSD was not a considered before.

Comment (safern) [reply eng/pipelines/coreclr/templates/build-job.yml:218]: Ah true.

**Pull request merged**